install_builder/deadline-cloud-for-cinema-4d.xml

```diff
@@ -0,0 +1,132 @@
+                </distributionDirectory>
+            </distributionFileList>
+        </folder>
+    </folderList>
+    <initializationActionList>
+        <if>
+            <conditionRuleList>
+                <compareText>
+                    <logic>does_not_contain</logic>
+                    <text>${platform_name}</text>
+                    <value>linux</value>
+                </compareText>
+            </conditionRuleList>
+            <actionList>
+                <setInstallerVariable name="all_components" value="${all_components} deadline_cloud_for_cinema_4d" />
+            </actionList>
+            <elseActionList>
+                <setInstallerVariable name="component(deadline_cloud_for_cinema_4d).show" value="0"/>
+            </elseActionList>
+        </if>
+        <setInstallerVariable name="cinema_4d_submitter_installdir" value="${installdir}\Submitters\Cinema4D" />
+        <if>
+            <conditionRuleList>
+                <platformTest type="windows" />
+            </conditionRuleList>
+            <actionList>
+                <setInstallerVariable name="cinema_4d_deps_platform" value="windows" />
+            </actionList>
+        </if>
+    </initializationActionList>
+    <parameterList>
+        <stringParameter name="deadline_cloud_for_cinema_4d_summary" ask="0" cliOptionShow="0">
+            <value>Deadline Cloud for Cinema 4D
+- Register the extension with Cinema 4D by moving the DeadlineCloud.pyp file to the Cinema 4D plugins folder. 
+- Set the C4DPYTHONPATH311 to retrieve deadline modules. 
+            </value>
+        </stringParameter>
+        <directoryParameter>
+            <name>cinema_4d_installdir</name>
+            <title>Cinema 4D</title>
+            <description>Cinema 4D Installation Directory</description>
+            <explanation>Path to Cinema 4D installation directory. For easiest installation, Cinema 4D should be installed before installing Deadline Cloud for Cinema 4D.</explanation>
+            <allowEmptyValue>0</allowEmptyValue>
+            <ask>yes</ask>
+            <cliOptionName>cinema4d-installation-directory</cliOptionName>
+            <cliOptionText>Path to Cinema 4D installation directory. </cliOptionText>
+            <mustBeWritable>yes</mustBeWritable>
+            <mustExist>1</mustExist>
+        </directoryParameter>
+    </parameterList>
+    <postInstallationActionList>
+        <createDirectory>
+            <path>${cinema_4d_installdir}/plugins</path>
+            <ruleList>
+                <fileTest condition="not_exists" path="${cinema_4d_installdir}/plugins"/>
+            </ruleList>
+        </createDirectory>
+        <copyFile>
+            <destination>${cinema_4d_installdir}/plugins/DeadlineCloud.pyp</destination>
+            <origin>${installdir}/tmp/c4d_deps/DeadlineCloud.pyp</origin>
+        </copyFile>
+        <showInfo>
+            <text>Moved DeadlineCloud.pyp to ${cinema_4d_installdir}/plugins/</text>
```

Comment:
Was this tested with a command line installation to see how it affects the flow there? I don't think it will block if running in unattended mode, but worth checking if you haven't already. Personally I'm not sure that this is needed as to me it just interrupts the installation flow with a popup that must be acknowledged I think?

Reply:
Yup, I tried just now with the CLI. 

```
D:\Users\pattatk\Downloads> DeadlineCloudSubmitter-windows-x64-installer.exe --mode unattended --enable-components deadline_cloud_for_cinema_4d  --cinema4d-plugins-directory "C:\Program Files\Maxon Cinema 4D 2025"
```

And the logs suggests that everything was installed as expected. 

```
...
Moved DeadlineCloud.pyp to C:\Program Files\Maxon Cinema 4D 2025/plugins/
Setting C4DPYTHONPATH311
Added C4DPYTHONPATH311 environment variable with value D:\Users\pattatk/DeadlineCloudSubmitter\Submitters\Cinema4D in scope user
Broadcasting environment changes
...
Creating Uninstaller
Creating uninstaller 25%
Creating uninstaller 50%
Creating uninstaller 75%
Creating uninstaller 100%
Uninstaller icon changed
Installation completed
Log finished 11/12/2024 at 15:04:51
Exiting with code 0

```